### PR TITLE
[TW-420] Update comments for errors 

### PR DESCRIPTION
**Pull request by kimberleehowley**
## What

This PR updates comments for errors, and adds an internal tag to what is maybe a helpful for a `toString` function. 

## Why

Improving the comments that will be used to generate documentation will help make the SDK easier to use. 

## Test Plan

- ❌ | No unit tests are needed for comment changes 
- ✅ | The reviewer should manually test the changes in this PR. 

Checkout my branch. 
Run: 

```
./gradlew dokkaHtml
npx http-serve reference-docs
```

### CODE REVIEW

File: forage-android/src/main/java/com/joinforage/forage/android/core/element/ElementValidationErrors.kt [hunk 1]
```diff
@@ -20,9 +38,17 @@ val TooLongEbtPanError = ElementValidationError(
...
+/**
+ * A type of [ElementValidationError] that is thrown when a customer submits an incomplete
+ * EBT Card PIN.
+ */
 val IncompleteEbtPinError = ElementValidationError(
     detail = "Your EBT card PIN is incomplete."
 )
+
+/**
+ * A type of [ElementValidationError] that is thrown when a customer submits an invalid EBT Card PIN.
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/core/element/ElementValidationErrors.kt:50]: the errors in this file are not exactly "thrown", but rather "set" (the `state.validationError` is set with the `ElementValidationError`)

File: forage-android/src/main/java/com/joinforage/forage/android/network/model/Message.kt [hunk 2]
```diff
@@ -2,9 +2,20 @@ package com.joinforage.forage.android.network.model
...
+    /**
+     * An error that is returned when a customer's EBT Card balance is insufficient to complete a
+     * payment.
+     * @property snapBalance A string that represents the available SNAP balance on the EBT Card.
+     * @property cashBalance A string that represents the available EBT Cash balance on the EBT Card.
+     * @see [Forage guide to handling insufficient funds](https://docs.joinforage.app/docs/plan-for-insufficient-ebt-funds-errors)
+     */
     data class EbtError51Details(val snapBalance: String? = null, val cashBalance: String? = null) : ForageErrorDetails() {
-        companion object {
+        internal companion object {
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/network/model/Message.kt:18]: nice!

**Review by djoksimo — CHANGES_REQUESTED** [review]

File: forage-android/src/main/java/com/joinforage/forage/android/ForageSDKInterface.kt [hunk 3]
```diff
@@ -5,9 +5,11 @@ import com.joinforage.forage.android.ui.ForagePANEditText
 import com.joinforage.forage.android.ui.ForagePINEditText
 
 /**
- * Exception thrown when attempting to call pass a reference to a
- * ForageElement without first setting its ForageConfig via
- * `.setForageConfig()`.
+ * An [Exception](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-exception/) thrown if a
+ * reference to a [ForageElement][com.joinforage.forage.android.ui.ForageElement] is passed to a
+ * function before [setForageConfig][com.joinforage.forage.android.ui.AbstractForageElement.setForageConfig]
+ * is called on the Element.
```
[the remark this answers is not in the document]
Comment (kimberleehowley) [reply forage-android/src/main/java/com/joinforage/forage/android/ForageSDKInterface.kt:11]: Yes! Sorry, I went with auto-complete, tried to move fast, and didn't spend enough time thinking about how we don't want abstractForageElement to be used 🤦

File: forage-android/src/main/java/com/joinforage/forage/android/core/element/ElementValidationErrors.kt (hunk 1 above)
Comment (kimberleehowley) [reply forage-android/src/main/java/com/joinforage/forage/android/core/element/ElementValidationErrors.kt:50]: Thanks for syncing IRL that I can just say occurs 💐

**Review by djoksimo — APPROVED** [review]

**Review by djoksimo — APPROVED** [review]

**Pull request merged**